caffe2/mkl/operators/local_response_normalization_op.cc

```diff
@@ -0,0 +1,86 @@
+#include "caffe2/core/context.h"
+#include "caffe2/core/operator.h"
+#include "caffe2/operators/local_response_normalization_op.h"
+
+#include "caffe2/utils/mkl_utils.h"
+
+#ifdef CAFFE2_HAS_MKL_DNN
+
+namespace caffe2 {
+namespace mkl {
+
+template <typename T>
+class MKLLRNOp final : public LRNOpBase<T, MKLContext>{
+
+  public:        
+    MKLLRNOp(const OperatorDef &operator_def, Workspace *ws)
+        : LRNOpBase<T, MKLContext>(operator_def, ws) {}
+
+    ~MKLLRNOp() {
+        if (workspace_buffer_ != NULL) {
+      dnnReleaseBuffer<T>(workspace_buffer_);
+      workspace_buffer_ = NULL;
+      }
+    }
+
+    bool RunOnDeviceWithOrderNCHW() override;
+    bool RunOnDeviceWithOrderNHWC() override;
+     
+private:
+    vector<TIndex> cached_input_dims_;
+    LayoutWrapper<T> workspace_layout_;
+    T *workspace_buffer_ = nullptr;
+    PrimitiveWrapper<T> primitive_;
+    MKLMemory<T> buffer_;
+    void* resources_[dnnResourceNumber] = {0};  
+};    
+
+template <>
+bool MKLLRNOp<float>::RunOnDeviceWithOrderNCHW() {
+    auto& X = OperatorBase::Input<MKLMemory<float>>(0);
+    MKLMemory<float>* Y = OperatorBase::Output<MKLMemory<float>>(0);
+
+    if (cached_input_dims_ != X.dims())  {
```

Comment:
ditto about changed layout while dimensions stay the same. Also I see this code path being repeated across operators. Have you thought about having a helper function / template / macro for this?